### PR TITLE
rmw_implementation: 1.0.1-1 in 'foxy/distribution.yaml' [bloom]

### PR DESCRIPTION
**Pull request by ahcorde**
Increasing version of package(s) in repository `rmw_implementation` to `1.0.1-1`:

- upstream repository: https://github.com/ros2/rmw_implementation.git
- release repository: https://github.com/ros2-gbp/rmw_implementation-release.git
- distro file: `foxy/distribution.yaml`
- bloom version: `0.10.0`
- previous version for package: `1.0.0-1`

## rmw_implementation

```
* Foxy updated maintainers (#158 <https://github.com/ros2/rmw_implementation/issues/158>)
* Contributors: Alejandro Hernández Cordero
```

## test_rmw_implementation

```
* Fixed rmw_create_node arguments
* Add fault injection tests to construction/destroy APIs.  (#144 <https://github.com/ros2/rmw_implementation/issues/144>)
* Fixed function rmw_create_node in tests
* Add tests bad type_support implementation (#152 <https://github.com/ros2/rmw_implementation/issues/152>)
* Add tests for localhost-only node creation (#150 <https://github.com/ros2/rmw_implementation/issues/150>)
* Use 10x the intraprocess delay to wait for sent requests. (#148 <https://github.com/ros2/rmw_implementation/issues/148>)
* Added rmw_wait, rmw_create_wait_set, and rmw_destroy_wait_set tests (#139 <https://github.com/ros2/rmw_implementation/issues/139>)
* Added rmw_service_server_is_available tests (#140 <https://github.com/ros2/rmw_implementation/issues/140>)
* Add tests service/client request/response with bad arguments (#141 <https://github.com/ros2/rmw_implementation/issues/141>)
* Fixed function rmw_create_node in tests
* Added rmw_wait, rmw_create_wait_set, and rmw_destroy_wait_set tests (#139 <https://github.com/ros2/rmw_implementation/issues/139>)
* Added test for rmw_get_serialized_message_size (#142 <https://github.com/ros2/rmw_implementation/issues/142>)
* Add service/client construction/destruction API test coverage. (#138 <https://github.com/ros2/rmw_implementation/issues/138>)
* Added rmw_publisher_allocation and rmw_subscription_allocation related tests (#137 <https://github.com/ros2/rmw_implementation/issues/137>)
* Add tests take serialized with info bad arguments (#130 <https://github.com/ros2/rmw_implementation/issues/130>)
* Add gid API test coverage. (#134 <https://github.com/ros2/rmw_implementation/issues/134>)
* Add tests take bad arguments  (#125 <https://github.com/ros2/rmw_implementation/issues/125>)
* Bump graph API test coverage. (#132 <https://github.com/ros2/rmw_implementation/issues/132>)
* Add tests take sequence serialized with bad arguments (#129 <https://github.com/ros2/rmw_implementation/issues/129>)
* Add tests take sequence + take sequence with bad arguments (#128 <https://github.com/ros2/rmw_implementation/issues/128>)
* Add tests take with info bad arguments (#126 <https://github.com/ros2/rmw_implementation/issues/126>)
* Add tests for non-implemented rmw_take_* functions (#131 <https://github.com/ros2/rmw_implementation/issues/131>)
* Add tests publish serialized bad arguments (#124 <https://github.com/ros2/rmw_implementation/issues/124>)
* Add tests publish bad arguments (#123 <https://github.com/ros2/rmw_implementation/issues/123>)
* Add tests non-implemented functions + loan bad arguments (#122 <https://github.com/ros2/rmw_implementation/issues/122>)
* Add missing empty topic name tests. (#136 <https://github.com/ros2/rmw_implementation/issues/136>)
* Add rmw_get_serialization_format() smoke test. (#133 <https://github.com/ros2/rmw_implementation/issues/133>)
* Complete publisher/subscription QoS query API test coverage. (#120 <https://github.com/ros2/rmw_implementation/issues/120>)
* Remove duplicate assertions (#121 <https://github.com/ros2/rmw_implementation/issues/121>)
* Add publisher/subscription matched count API test coverage. (#119 <https://github.com/ros2/rmw_implementation/issues/119>)
* Fixed rmw_create_node
* Add serialize/deserialize API test coverage. (#118 <https://github.com/ros2/rmw_implementation/issues/118>)
* Add subscription API test coverage. (#117 <https://github.com/ros2/rmw_implementation/issues/117>)
* Extend publisher API test coverage (#115 <https://github.com/ros2/rmw_implementation/issues/115>)
* Add node construction/destruction API test coverage. (#112 <https://github.com/ros2/rmw_implementation/issues/112>)
* Check that rmw_init() fails if no enclave is given. (#113 <https://github.com/ros2/rmw_implementation/issues/113>)
* Add init options API test coverage. (#108 <https://github.com/ros2/rmw_implementation/issues/108>)
* Complete init/shutdown API test coverage. (#107 <https://github.com/ros2/rmw_implementation/issues/107>)
* Add dependency on ament_cmake_gtest (#109 <https://github.com/ros2/rmw_implementation/issues/109>)
* Add test_rmw_implementation package. (#106 <https://github.com/ros2/rmw_implementation/issues/106>)
* Contributors: Alejandro Hernández Cordero, Geoffrey Biggs, Jose Tomas Lorente, Michel Hidalgo, Shane Loretz
```

### CODE REVIEW

File: foxy/distribution.yaml [hunk 1]
```diff
@@ -2519,10 +2519,13 @@ repositories:
       url: https://github.com/ros2/rmw_implementation.git
       version: foxy
     release:
+      packages:
+      - rmw_implementation
+      - test_rmw_implementation
```
Comment (dirk-thomas) [inline foxy/distribution.yaml:2524]: Should this package really be released?

Comment (ahcorde) [reply foxy/distribution.yaml:2524]: It's just a package that contains tests. feel free to close it if this is not relevant or not adding any value.

I was following the steps done in rolling (where is [released]( https://github.com/ros/rosdistro/blob/master/rolling/distribution.yaml#L1629))

Comment (dirk-thomas) [reply foxy/distribution.yaml:2524]: > I was following the steps done in rolling (where is released)

This was added in #25720. @ivanpauno Can you clarify if that package needs to be made available from Debian or if it only contains tests and should be ignored in the gbp repo instead?

Comment (ivanpauno) [reply foxy/distribution.yaml:2524]: I'm not sure what's the procedure to only release some packages in a repository and not all.
I agree that `test_rmw_implementation` don't need to have Debian package.

Comment (dirk-thomas) [reply foxy/distribution.yaml:2524]: > I'm not sure what's the procedure to only release some packages in a repository and not all.

https://answers.ros.org/question/66545/to-ignore-some-pkgs-upon-release-by-bloom/

@ivanpauno The `.ignore` should be added for rolling and the `test_rmw_implementation` name should be removed from the rolling distribution.
@ahcorde The `test_rmw_implementation` name should be removed from this PR as well. No need to run bloom again - the already generated data for that package will simply not be used.

Comment (ivanpauno) [reply foxy/distribution.yaml:2524]: > @ivanpauno The .ignore should be added for rolling

Done in https://github.com/ros2-gbp/rmw_implementation-release/commit/d3cbd4f7ddaf407369733b447e9cbbd42864cbd9#diff-d1bed8395d4099eceeaaba8471aa916b025de36ae35f0993edb489af258f8df7.

Comment (dirk-thomas) [reply foxy/distribution.yaml]: @ivan Please still remove the package name from the rolling distribution file.

**Comment by dirk-thomas** [comment]
@ahcorde Please see the failing CI for what needs to be changed.

File: foxy/distribution.yaml (hunk 1 above)
Comment (ivanpauno) [reply foxy/distribution.yaml]: > Please still remove the package name from the rolling distribution file.

Ah, I thought this PR was doing that. See https://github.com/ros/rosdistro/pull/27104.

Comment (ivanpauno) [reply foxy/distribution.yaml]: Considering this PR, should a `foxy.ignored` file be added too?

**Pull request merged**